streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindows.java

```diff
@@ -83,16 +92,52 @@ private TimeWindows(final long sizeMs, final long advanceMs, final long graceMs)
      * Tumbling windows are a special case of hopping windows with {@code advance == size}.
      *
      * @param size The size of the window
-     * @return a new window definition with default maintain duration of 1 day
+     * @return a new window definition with default no grace period
+     * @throws IllegalArgumentException if the specified window size is zero or negative or can't be represented as {@code long milliseconds}
+     */
+    public static TimeWindows ofSizeWithNoGrace(final Duration size) throws IllegalArgumentException {
```

Comment:
We should call out explicitly that this is setting the grace period to 0, which means that out of order records arriving after the window end will be dropped. Otherwise it's too easy to just use this method without thinking any further about the grace period and what it means/whether you want it

Reply:
Call it out explicitly in the javadocs, I mean. ie in the description part, not just the `@return` (though it's good to have put it there too). It would be a good idea to describe what the grace period actually means in the javadocs of this method like we do in `ofSizeAndGrace` so that users know what it is they are losing by opting out of the grace period.

Reply:
In fact, since this and `ofSizeAndGrace` are pretty much the same thing -- differing only in the value of one config -- they should have the same javadocs completely. 

Only the `@returns` and `@param` would be different. And maybe one sentence in each which says either `"this window definition allows you to specify the grace period"` or `"this window definition does not require setting the grace period, note that this means out-of-order data arriving after the window end will be dropped so think carefully about your requirements before using this in production"` --- or something like that

Reply:
@ableegoldman 

Makes sense. I will review the Javadocs again for all the API changes and apply this feedback to the new API methods for the KIP